tests/k8s/test_k8s.py

```diff
@@ -587,3 +589,58 @@ async def test_flow_connection_pool(logger, k8s_connection_pool, docker_images,
         assert len(visited) == 2
     else:
         assert len(visited) == 1
+
+
+@pytest.mark.asyncio
+@pytest.mark.timeout(3600)
+@pytest.mark.parametrize(
+    'docker_images',
+    [['jinaai/jina']],
+    indirect=True,
+)
+async def test_flow_with_external_pod(logger, docker_images, tmpdir):
+    class DocReplaceExecutor(Executor):
```

Comment:
just for completion, can we add a test where the external Pod is in Kubernetes?

Reply:
I could add this, but I am not sure its providing much value. This test would actually test more the external pod itself being in K8s which is not what this feature is about (its about the Flow using a given external Pod). 

From the Flow/Gateway perspective it should not matter whatsever where the external Pod is located, we just connect to any given address.

Reply:
It would document, ensure that the pattern works